bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -803,8 +885,72 @@ void checkAllLedgers() throws BKException, IOException, InterruptedException, Ke
                 LOG.error("Got exception while trying to set checkAllLedgersCTime", ue);
             }
         } finally {
-            admin.close();
-            client.close();
+            localAdmin.close();
+            localClient.close();
+        }
+    }
+
+    void metadataCheck() throws BKAuditException {
+        final CountDownLatch metadataCheckLatch = new CountDownLatch(1);
+        Processor<Long> ledgerProcessor = new Processor<Long>() {
+            @Override
+            public void process(Long ledgerId, AsyncCallback.VoidCallback iterCallback) {
+                ledgerManager.readLedgerMetadata(ledgerId).whenComplete((metadataVer, exception) -> {
+                    if (exception == null) {
+                        LedgerMetadata metadata = metadataVer.getValue();
+                        int writeQuorumSize = metadata.getWriteQuorumSize();
+                        int ackQuorumSize = metadata.getAckQuorumSize();
+                        if (metadata.isClosed()) {
```

Comment:
Ah good you are just checking closed ledgers; the above (fence) is just a comment issue; please take care of the comment.

Reply:
ok will change comment to 'closed ledgers'